src/main/java/org/opensearch/geospatial/index/query/xyshape/XYShapeQueryVisitor.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.query.xyshape;
+
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYCircle;
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYLine;
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYPoint;
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYPolygon;
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYRectangle;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.lucene.geo.XYGeometry;
+import org.opensearch.geometry.Circle;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.GeometryCollection;
+import org.opensearch.geometry.GeometryVisitor;
+import org.opensearch.geometry.Line;
+import org.opensearch.geometry.LinearRing;
+import org.opensearch.geometry.MultiLine;
+import org.opensearch.geometry.MultiPoint;
+import org.opensearch.geometry.MultiPolygon;
+import org.opensearch.geometry.Point;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geometry.ShapeType;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.QueryShardException;
+
+/**
+ * Geomtery Visitor to create a query to find all cartesian XYShapes
```

Comment:
nit: typo "Geomtery" should be "Geometry"

Reply:
Ack